openedx/features/course_duration_limits/access.py

```diff
@@ -115,8 +115,8 @@ def register_course_expired_message(request, course):
     if CourseDurationLimitConfig.enabled_for_enrollment(user=request.user, course_key=course.id):
         expiration_date = get_user_course_expiration_date(request.user, course)
         if expiration_date:
-            upgrade_message = _('Your access to this course expires on {expiration_date}. \
-                    <a href="{upgrade_link}">Upgrade now</a> for unlimited access.')
+            upgrade_message = _('<div>Your access to this course expires on {expiration_date}. \
+                    <a href="{upgrade_link}">Upgrade now <span class="sr-only">to retain access past {expiration_date}.</span></a><span aria-hidden="true">for unlimited access.</span></div>')
```

Comment:
The banner here already has role="alert". In theory, role="alert" should cause this message to get read everytime the page loads. In practice, I am not hearing this automatically read as an alert with VoiceOver/Safari (the more official combination). I do see this reliably read on page load with Voice Over/Chrome on course home, but not in the courseware.  
Our goal with this banner is to ensure users are aware of it. We also may not want to be overly disruptive.  
I think this solution may be a good solution. When users read through the whole page, or browse through the links menu, or tab through the links they will see this banner. By adding screenreader text to the link, users will be aware of the course expiration when browsing through the links menu. IMO this may be better than something like moving focus to banner at each page load which will be more disruptive than helpful after a user listens to the banner a few times.